compiler/rustc_target/src/abi/call/sparc64.rs

```diff
@@ -35,12 +35,14 @@ where
         data.last_offset = data.last_offset + Reg::i32().size;
     }
 
-    for _ in 0..((offset - data.last_offset).bits() / 64)
-        .min((data.prefix.len() - data.prefix_index) as u64)
-    {
-        data.prefix[data.prefix_index] = Some(Reg::i64());
-        data.prefix_index += 1;
-        data.last_offset = data.last_offset + Reg::i64().size;
+    if data.last_offset < offset {
+        for _ in 0..((offset - data.last_offset).bits() / 64)
+            .min((data.prefix.len() - data.prefix_index) as u64)
+        {
+            data.prefix[data.prefix_index] = Some(Reg::i64());
+            data.prefix_index += 1;
+            data.last_offset = data.last_offset + Reg::i64().size;
+        }
     }
```

Comment:
I believe this new `if` can never change behavior, except in cases which would panic before.

Going case-by-case:
- `data.last_offset < offset`: we enter the loop, as before
- `data.last_offset == offset`: after this change, we skip the loop; before this change, `offset - data.last_offset` is zero, so the loop iterates over `0..0`, so the loop executes zero iterations, which is equivalent
- `data.last_offset > offset`: after this change, we skip the loop; before this change `offset - data.last_offset` panics due to overflow